### PR TITLE
mrsal-21: Test using NOX and create test with SSL connection 

### PR DESCRIPTION
**Pull request by rafatzahran**
Investigate & use nox.
Add new test with SSL/TLS connection.

### CODE REVIEW

**Review by MartinRovang — APPROVED** [review]
Good amount of tests :D

File: tests/test_ssl/test_with_ssl.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+              virtual_host=virtual_host,
+              ssl=True)
+
+print(host)
+print(port)
+print(credentials)
+print(virtual_host)
+print(os.environ.get('RABBITMQ_CAFILE'))
+print(os.environ.get('RABBITMQ_CERT'))
+print(os.environ.get('RABBITMQ_KEY'))
```
Comment (JonNesvold) [inline tests/test_ssl/test_with_ssl.py:30]: are these necessary?

File: tests/test_ssl/test_with_ssl.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+host = os.environ.get('RABBITMQ_DOMAIN_TLS')
+port = os.environ.get('RABBITMQ_PORT_TLS')
+credentials = (os.environ.get('RABBITMQ_DEFAULT_USER'), os.environ.get('RABBITMQ_DEFAULT_PASS'))
+virtual_host = os.environ.get('RABBITMQ_DEFAULT_VHOST')
+
+mrsal = Mrsal(host=host,
+              port=port,
+              credentials=credentials,
+              virtual_host=virtual_host,
+              ssl=True)
```
Comment (JonNesvold) [inline tests/test_ssl/test_with_ssl.py:22]: you need to add in docs how to test this. We can't publish our credentials or certificates. Let the user know that they need to give the path to the TLS certs they have created and if they don't do this then SSL setup will fail.

**Review by JonNesvold — APPROVED** [review]
Nice clean up and well done. See my comments about SSL testing and consider using verbose with logs we are printing from Mrsal. I think it is very individual how people like their logs. Too much logs is not good and too little is also not good, hence verbose with even levels would be a nice feature.

Merge when comments are considered and accounted for.

**Comment by JonNesvold** [comment]
Add coverage badge to the repo. It can be done in a bash script which uses the xml files generated by the noxfile. A pacakge called genbadges or copy pase Martin's bash script. Add the badge file automatically in GitHub workflow.  Just specifiy the path.

**Review by JonNesvold — APPROVED** [review]

**Pull request merged**